### PR TITLE
Mikes blend/add list to secrets

### PR DESCRIPTION
**Pull request by mikes-blend**
Add a "List" operation to the secrets interface

### CODE REVIEW

File: request/constants.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ const (
 	MethodDelete = "DELETE"
 	// MethodOptions is a method.
 	MethodOptions = "OPTIONS"
+	// MethodList is a method.
+	MethodList = "LIST"
```
Comment (mikes-blend) [inline request/constants.go]: Can take this out if you want. These aren't the constants that secrets uses

File: secrets/mock_client.go [hunk 2]
```diff
@@ -45,3 +46,28 @@ func (c *MockClient) Delete(_ context.Context, key string, options ...Option) er
 	delete(c.SecretValues, key)
 	return nil
 }
+
+// List lists keys on a path
+func (c *MockClient) List(_ context.Context, path string, options ...Option) ([]string, error) {
```
Comment (mikes-blend) [inline secrets/mock_client.go]: Was harder to mock than implement 🙄

File: request/constants.go (hunk 1 above)
Comment (wcharczuk) [reply request/constants.go]: AFAIK it's not a standard HTTP verb, I would omit.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods

File: secrets/mock_client.go (hunk 2 above)
Comment (wcharczuk) [reply secrets/mock_client.go]: Yikes no kidding

File: request/constants.go (hunk 1 above)
Comment (mikes-blend) [reply request/constants.go]: I figured it was in the extended registry, but it is not. Can’t believe hashicorp just made their own shit up.

http://www.iana.org/assignments/http-methods/http-methods.xhtml

**Pull request merged**